samples/philosophers/sample.yaml

```diff
@@ -36,6 +36,7 @@ tests:
       - CONFIG_NUM_PREEMPT_PRIORITIES=6
       - CONFIG_NUM_COOP_PRIORITIES=0
       - CONFIG_BT=n
+      - CONFIG_NETWORKING=n
```

Comment:
This certainly works but I am wondering if we should fix this in Kconfig level.
The "Too many traffic classes" message is shown as we have this check in the code
```
BUILD_ASSERT_MSG(NET_TC_RX_COUNT <= CONFIG_NUM_COOP_PRIORITIES,
			 "Too many traffic classes");
```
and for this test we set `CONFIG_NUM_COOP_PRIORITIES=0` which then leads to the build failure.

Probably better fix would be to have dependency in `subsys/net/ip/Kconfig` and some checks like this
```
diff --git a/subsys/net/ip/Kconfig b/subsys/net/ip/Kconfig
index 7ebc95eceb..8553ea7ad7 100644
--- a/subsys/net/ip/Kconfig
+++ b/subsys/net/ip/Kconfig
@@ -129,6 +129,7 @@ config NET_TC_TX_COUNT
 	int "How many Tx traffic classes to have for each network device"
 	default 1
 	range 1 8
+	depends on NUM_COOP_PRIORITIES > 0
 	help
 	  Define how many Tx traffic classes (queues) the system should have
 	  when sending a network packet. The network packet priority can then
@@ -144,6 +145,7 @@ config NET_TC_RX_COUNT
 	int "How many Rx traffic classes to have for each network device"
 	default 1
 	range 1 8
+	depends on NUM_COOP_PRIORITIES > 0
 	help
 	  Define how many Rx traffic classes (queues) the system should have
 	  when receiving a network packet. The network packet priority can then
diff --git a/subsys/net/ip/net_tc.c b/subsys/net/ip/net_tc.c
index b136f53ec6..1f56ef0ca2 100644
--- a/subsys/net/ip/net_tc.c
+++ b/subsys/net/ip/net_tc.c
@@ -108,8 +108,10 @@ static u8_t tx_tc2thread(u8_t tc)
 #endif
 	};
 
+#if CONFIG_NUM_COOP_PRIORITIES > 0
 	BUILD_ASSERT_MSG(NET_TC_TX_COUNT <= CONFIG_NUM_COOP_PRIORITIES,
 			 "Too many traffic classes");
+#endif
 
 	NET_ASSERT(tc < ARRAY_SIZE(thread_priorities));
 
@@ -161,8 +163,10 @@ static u8_t rx_tc2thread(u8_t tc)
 #endif
 	};
 
+#if CONFIG_NUM_COOP_PRIORITIES > 0
 	BUILD_ASSERT_MSG(NET_TC_RX_COUNT <= CONFIG_NUM_COOP_PRIORITIES,
 			 "Too many traffic classes");
+#endif
 
 	NET_ASSERT(tc < ARRAY_SIZE(thread_priorities));
```

Reply:
Well, this change was obviously just following the pattern of `CONFIG_BT=n`. And as I agree that these changes should not be required, I am a bit worried of opening the Pandora's box - `CONFIG_NUM_COOP_PRIORITIES` is used in quite a few places in runtime without any sanity checks.

Ifdefing this check does not lead to a compile-time crash, but will `net_tc_tx_init` work fine with `NUM_COOP_PRIORITIES` equal to 0?

And in fact, shouldn't we consider using `CONFIG_COOP_ENABLED=n` in this test? It doesn't solve anything as of now, as the code you ifdefed does not take it into account, but maybe it should?

What do you think @jukkar? I'm worried that this can escalate very quickly, but Zephyr's scheduling is not my primary field of expertise here, so I'm eager to hear from you.